### PR TITLE
Add .NET Core support

### PR DESCRIPTION
**Pull request by lukespice**
I want to use this in a .NET Core project so I have converted AvnAs.csproj to .NET Standard 1.4.
I have built the NuGet package locally and it works for me.

57 tests pass and 8 fail, that is the same as was running before the conversion.

### CODE REVIEW

**Comment by EamonNerbonne** [comment]
Cool!

Sorry about the test failures (can you be more specific?)  I suspect that's due to the approval tests?  I occasionally update the dataset, and if I feel the new dataset does better on the approvals than the old, I update and re-approve.  The released version should thus not have approval failures, but I've been known to be a little lazy...  The "correctness" of the approvals (given the problem domain) is rather fuzzy; I'm still using normal testing frameworks because that's simpler.

---

I'll definitely use this and upgrade to .net core eventually, but I'm going to brainstorm aloud some of the tradeoffs as to when exactly this should become the primary version on nuget.

 - Unfortunately, at the moment you cannot easily be .net core and .net framework compatible without requiring .net 461.  I feel that's a slightly high bar.  Rumor has it that .net standard 2.0 may  *reduce* the .net framework requirement, so waiting may help here
 - Multitargeting is poorly documented, but surprisingly simple `<TargetFramework>` becomes `<TargetFrameworks>` and you use a semi-colon separated list.  That might be a workaround, but it's also work.  Also, visual studio doesn't really do much with non-primary frameworks IIRC.  Since apparently no source changes were necessary we can get away without preprocessor symbols!
 - The msbuild-integrated nuget does not support all nuget fields (oddly enough!)  In particular, it merges description and summary.  That might be OK; it's not ideal.  For another project I tried using external nuget, but the documentation was still very poor; I couldn't get everything working together smoothly.


So; three choices:

 - accept a high framework requirement (net461) and break backwards compatibility.
 - wait for .net standard 2.0
 - multitargetting.
 - ...anything else?

What do you think?

File: A-vs-An/A-vs-An-DotNet/AvsAn.csproj [hunk 1]
```diff
@@ -1,62 +1,15 @@
...
-    <AssemblyName>AvsAn</AssemblyName>
-    <TargetFrameworkVersion>v2.0</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
-    <TargetFrameworkProfile>
-    </TargetFrameworkProfile>
+    <TargetFramework>netstandard1.4</TargetFramework>
+    <Company>Eamon Nerbonne</Company>
+    <Description>Find the english language indeterminate article ('a' or 'an') for a word.  Based on real usage patterns extracted from the wikipedia text dump; can therefore even deal with tricky edge cases such as acronyms (FIAT vs. FAA, NASA vs. NSA) and odd symbols.  (Requires .NET Core or .NET 4.6.1)</Description>
+    <Version>2.3.0</Version>
+    <Copyright>Copyright Eamon Nerbonne ©  2014</Copyright>
```
Comment (EamonNerbonne) [inline A-vs-An/A-vs-An-DotNet/AvsAn.csproj]: See https://docs.microsoft.com/en-us/nuget/schema/msbuild-targets for some fields missing from the nuget conversion.

File: A-vs-An/AvsAn-JsDemo/AvsAn-JsDemo.csproj [hunk 2]
```diff
@@ -45,6 +45,11 @@
       <HintPath>..\packages\Microsoft.Web.Infrastructure.1.0.0.0\lib\net40\Microsoft.Web.Infrastructure.dll</HintPath>
     </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Data.DataSetExtensions" />
+    <Reference Include="System.Web.ApplicationServices" />
+    <Reference Include="System.Web.DynamicData" />
+    <Reference Include="System.Web.Entity" />
+    <Reference Include="System.Web.Extensions" />
```
Comment (EamonNerbonne) [inline A-vs-An/AvsAn-JsDemo/AvsAn-JsDemo.csproj]: Not that it really matters, but is this necessary?

File: A-vs-An/A-vs-An-DotNet/AvsAn.nuspec [hunk 3]
```diff
@@ -12,14 +12,10 @@
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <summary>$description$</summary>
     <description>
-      AvsAn: Find the english language indeterminate article ("a" or "an") for a word.  Based on real usage patterns extracted from the wikipedia text dump; can therefore even deal with tricky edge cases such as acronyms (FIAT vs. FAA, NASA vs. NSA) and odd symbols.  (Requires .NET 2.0 or higher)
+      AvsAn: Find the english language indeterminate article ("a" or "an") for a word.  Based on real usage patterns extracted from the wikipedia text dump; can therefore even deal with tricky edge cases such as acronyms (FIAT vs. FAA, NASA vs. NSA) and odd symbols.  (Requires .NET Core or .NET 4.6.1)
```
Comment (EamonNerbonne) [inline A-vs-An/A-vs-An-DotNet/AvsAn.nuspec]: .net core 1.0 or 1.1?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# User-specific files
+*.suo
+*.user
+
+# Build results
+[Bb]in/
+[Oo]bj/
+*.csproj.user
+*.pubxml.user
```
Comment (EamonNerbonne) [inline .gitignore]: redundant.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+A-vs-An/PerfTest/354984si.ngl
+
+*.vsp
+*.psess
+*.suo
+*.vspx
+packages/
+*.DotSettings.user
```
Comment (EamonNerbonne) [inline .gitignore]: redundant

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+*.nupkg
+*.received.txt
+*.approved.txt.bak
```
Comment (EamonNerbonne) [inline .gitignore]: might as wel *.bak

File: A-vs-An/A-vs-An-DotNet/AvsAn.csproj (hunk 1 above)
Comment (EamonNerbonne) [inline A-vs-An/A-vs-An-DotNet/AvsAn.csproj]: Feel free to add yourself! :-)

File: .gitignore (hunk 6 above)
Comment (lukespice) [inline .gitignore]: I copied the .hgignore and made some changes changes to the format to work for me. I typically use https://www.gitignore.io/api/visualstudio in my own projects, it's pretty massive though and has a lot of redundant lines, but makes life simple.

File: A-vs-An/A-vs-An-DotNet/AvsAn.nuspec (hunk 3 above)
Comment (lukespice) [inline A-vs-An/A-vs-An-DotNet/AvsAn.nuspec]: I think either work, but I only tested it with my .net 1.1 project.

File: A-vs-An/AvsAn-JsDemo/AvsAn-JsDemo.csproj (hunk 2 above)
Comment (lukespice) [inline A-vs-An/AvsAn-JsDemo/AvsAn-JsDemo.csproj]: Visual Studio must have snuck that in, I will remove.

**Comment by lukespice** [comment]
Thanks for taking the time to review, these are the failing tests.

WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_AaRiver
WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_Abatement
WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_AfrikaIslam
WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_CookIslandsTransport
WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_Disperser
WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_DominicaTransport
WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_FeaturedList
WikipediaAvsAnTrieExtractorTest.MarkupStripperTests.Wikipage_Glottis

I agree the .NET Standard targeting is a bit weird and restrictive at the moment. There is nothing in the code that needs 4.6.1 so it shouldn't be a requirement. But if that's the only current way to distribute to .NET core at the moment maybe it's worth the trade-off?

Or happy to wait for .NET Standard 2.0 and see how that works out.

**Comment by EamonNerbonne** [comment]
For some reason I missed this github notification; sorry!

**Comment by EamonNerbonne** [comment]
Thanks for all the timeconsuming conversion work!

**Comment by EamonNerbonne** [comment]
If you have any further changes you want to make; feel free to push those into the repo.

I'll try and get around to a multi-targeted nuget package asap, or maybe I'll just bump the major version number and accept the relatively high .net version requirement.  That's certainly simpler...

**Comment by EamonNerbonne** [comment]
Oh, and to be explicit: can you confirm you saw the apache license at https://github.com/EamonNerbonne/a-vs-an/blob/master/LICENSE and that you license any code you commit to this repo under that license?

**Comment by lukespice** [comment]
Of course the license is fine. Thanks for the merge!

**Comment by EamonNerbonne** [comment]
I published this as version 3.0.0 to nuget.  I decided to bump the major version because it's a breaking change for anyone running .net 2.0 up to and excluding 4.5; even though the number of people that applies to that use nuget must be approaching 0 these days.

**Comment by lukespice** [comment]
Awesome, nice work. The older versions will still be available for anyone on a lower .NET version.

**Pull request merged**